src/analysis/retail/evoker/preservation/modules/talents/TitansGift.tsx

```diff
@@ -0,0 +1,182 @@
+import Analyzer, { Options, SELECTED_PLAYER } from 'parser/core/Analyzer';
+import { TALENTS_EVOKER } from 'common/TALENTS';
+import Statistic from 'parser/ui/Statistic';
+import STATISTIC_ORDER from 'parser/ui/STATISTIC_ORDER';
+import STATISTIC_CATEGORY from 'parser/ui/STATISTIC_CATEGORY';
+import ItemHealingDone from 'parser/ui/ItemHealingDone';
+import { SpellLink, TooltipElement } from 'interface';
+import SPELLS from 'common/SPELLS';
+import Events, { ApplyBuffEvent, EventType, HealEvent, RefreshBuffEvent } from 'parser/core/Events';
+import {
+  getBlossomCast,
+  getDreamBreathHealing,
+  getReversionHealing,
+  getEchoAplication,
+  getHealEvents,
+  isCastFromBurst,
+} from '../../normalizers/EventLinking/helpers';
+import { calculateEffectiveHealing } from 'parser/core/EventCalculateLib';
+import { TITANS_GIFT_INC } from '../../normalizers/EventLinking/constants';
+
+class TitansGift extends Analyzer {
+  //Blossom
+  healingAddedToBlossoms: number = 0;
+  totalBlossomsCasted: number = 0;
+  buffedBlossoms: number = 0;
+  //Echo
+  healingAddedToEcho: number = 0;
+  totalEchoesCasted: number = 0;
+  buffedEchoes: number = 0;
+
+  constructor(options: Options) {
+    super(options);
+
+    this.active = this.selectedCombatant.hasTalent(TALENTS_EVOKER.TITANS_GIFT_TALENT);
+
+    //Count total casts
+    this.addEventListener(
+      Events.cast.by(SELECTED_PLAYER).spell(SPELLS.EMERALD_BLOSSOM_CAST),
+      this.onBlossomCasts,
+    );
+
+    this.addEventListener(
+      Events.cast.by(SELECTED_PLAYER).spell(TALENTS_EVOKER.ECHO_TALENT),
+      this.onEchoCasts,
+    );
+
+    //Track blossom heals
+    this.addEventListener(
+      Events.heal.by(SELECTED_PLAYER).spell(SPELLS.EMERALD_BLOSSOM),
+      this.emeraldBlossomHeal,
+    );
+
+    //Track echo heals
+    this.addEventListener(
+      Events.heal
+        .by(SELECTED_PLAYER)
+        .spell([
+          SPELLS.DREAM_BREATH_ECHO,
+          SPELLS.EMERALD_BLOSSOM_ECHO,
+          SPELLS.LIVING_FLAME_HEAL,
+          SPELLS.SPIRITBLOOM_SPLIT,
+          SPELLS.SPIRITBLOOM_FONT,
+          SPELLS.SPIRITBLOOM,
+          SPELLS.VERDANT_EMBRACE_HEAL,
```

Comment:
you can use `constants.ECHO_HEALS`

Reply:
i can't get the `buffedEchoes` amout to match the amount of echoes casted with essence burst changing this, im using the same list thats used in the `ECHO` link in `EchoEventLinks.ts`